pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -28,31 +33,130 @@
+    """
+    schema = schema.copy()
+    schema_update, _ = collect_known_metadata([annotation])
+    if isinstance(annotation, at.Gt):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.greater_than_validator, gt=annotation.gt),
+                schema,
+            )
+    elif isinstance(annotation, at.Ge):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.greater_than_or_equal_validator, ge=annotation.ge),
+                schema,
+            )
+    elif isinstance(annotation, at.Lt):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.less_than_validator, lt=annotation.lt),
+                schema,
+            )
+    elif isinstance(annotation, at.Le):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.less_than_or_equal_validator, le=annotation.le),
+                schema,
+            )
+    elif isinstance(annotation, at.MultipleOf):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.multiple_of_validator, multiple_of=annotation.multiple_of),
+                schema,
+            )
+    elif isinstance(annotation, at.MinLen):
+        if schema['type'] in SEQUENCE_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.min_length_validator, min_length=annotation.min_length),
+                schema,
+            )
+    elif isinstance(annotation, at.MaxLen):
+        if schema['type'] in SEQUENCE_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.max_length_validator, max_length=annotation.max_length),
+                schema,
+            )
+    elif isinstance(annotation, at.Predicate):
+        predicate_name = f'{annotation.func.__qualname__} ' if hasattr(annotation.func, '__qualname__') else ''
+
+        def val_func(v: Any) -> Any:
+            # annotation.func may also raise an exception, let it pass through
+            assert annotation.func(v), f'Predicate {predicate_name}failed'
```

Comment:
might be cleaner to raise a `PydanticCustomError`.

Reply:
Any suggestions for the right error to raise? I was hoping to make it "similar" to what would happen if you called the predicate by hand inside of your own functional validator, and since the predicate returns a boolean I felt like an `assert` would be the natural thing to do.

Reply:
done